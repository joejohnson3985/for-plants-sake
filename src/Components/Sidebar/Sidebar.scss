@import '../../Utilities/_Variables.scss';

i {
  font-size: 30px;
  margin: 25px;
  color: $dark-accent;

  &:hover {
    color: $light-accent;
    cursor: pointer;
  }
}

.side-bar-container {
  position: absolute;
}


.bar {
  background: $light-text;
  text-align: center;
  width: 0;
  top: 0;
  left: 0;
  overflow-x: hidden;
  transition: 0.5s;
  position: fixed;
  box-shadow: 0 0 11px rgba(33,33,33,.6);
  color: $light-text;

  i {
    position: absolute;
    top: 0;
    right: 0;
  }
  

  h1 {
    font-size: 30px;
    margin: 0;
  }
}

.show-nav {
  width: 250px;
  padding: 20px;
  margin: 15px 0 0 15px;
  z-index: 10;
  color: $primary;
}

.nav-link {
  color: $primary;
  text-decoration: none;
  font-size: 24px;

  &:hover {
    color: $secondary;
  }
}

.current-nav {
  color: $secondary;
}

