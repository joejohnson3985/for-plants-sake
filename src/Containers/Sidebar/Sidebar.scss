@import '../../Utilities/_Variables.scss';

i {
  font-size: 30px;
  margin: 25px;
  color: $dark-accent;

  &:hover {
    color: $light-accent;
    cursor: pointer;
  }
}

.side-bar-container {
  position: absolute;
}


.bar {
  background: $light-text;
  color: $dark-accent;
  text-align: center;
  width: 0;
  z-index: 1;
  top: 0;
  left: 0;
  overflow-x: hidden;
  transition: 0.7s;
  position: fixed;
  box-shadow: 0 0 11px rgba(33,33,33,.6);

  i {
    position: absolute;
    top: 0;
    right: 0;
  }
  

  h1 {
    font-size: 30px;
    margin: 0;
  }
}

.show-nav {
  width: 250px;
  padding: 20px;
  margin: 15px 0 0 15px;
}