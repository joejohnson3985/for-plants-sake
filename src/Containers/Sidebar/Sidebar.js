import React, { Component } from 'react';
import './Sidebar.scss';
import { NavLink } from 'react-router-dom';

class Sidebar extends Component {
  constructor() {
    super()
    this.state = {
      showNav: false
    }
  }

  showSidebar = () => {
    this.setState({showNav: !this.state.showNav})
  }

  render() {
    let css = 'bar';
    if(this.state.showNav) {css = 'bar show-nav' }
    return(
      <div className='side-bar-container'>
        <i className='fas fa-bars' onClick={this.showSidebar}></i>
        <div className={css}>
          <div>
            <i className='fas fa-times' onClick={this.showSidebar}></i>
            <h1>Peaking</h1>
            <NavLink exact={true} to="/search" activeClassName='current-nav' className='nav-link' onClick={this.showSidebar} ><h4>Find Hikes</h4></NavLink>
            <NavLink exact={true} to="/user/future-hikes" activeClassName='current-nav' className='nav-link' onClick={this.showSidebar} ><h4>Future Hikes</h4></NavLink>
            <NavLink exact={true} to="/user/completed-hikes" activeClassName='current-nav' className='nav-link' onClick={this.showSidebar} ><h4>Completed Hikes</h4></NavLink>
          </div>
        </div>
      </div>
    )
  }
}

export default Sidebar;
