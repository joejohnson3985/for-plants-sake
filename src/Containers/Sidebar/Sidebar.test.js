import Sidebar from './Sidebar';
import { shallow } from 'enzyme';
import React from 'react';


const store = jest.fn()
let wrapper;

describe('Sidebar', () => {

  beforeEach(() => {
    wrapper = shallow(
      <Sidebar />
    )
  })
  it('Should match the snapshot when showNav is false', () => {
    expect(wrapper.state().showNav).toEqual(false)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot when showNav is true', () => {
    wrapper.state().showNav = true
    expect(wrapper.state().showNav).toEqual(true)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should have default state', () => {
    expect(wrapper.state()).toEqual({showNav: false})
  })

  it('Should call showSidebar on click of find hikes and update state', () => {
    expect(wrapper.state().showNav).toEqual(false)
    jest.spyOn(wrapper.instance(), 'showSidebar')
    wrapper.find('.find-hikes-link').simulate('click')
    expect(wrapper.instance().showSidebar).toHaveBeenCalled()
    expect(wrapper.state().showNav).toEqual(true)
  })

  it('Should call showSidebar on click of my hikes and update state', () => {
    expect(wrapper.state().showNav).toEqual(false)
    jest.spyOn(wrapper.instance(), 'showSidebar')
    wrapper.find('.my-hikes-link').simulate('click')
    expect(wrapper.instance().showSidebar).toHaveBeenCalled()
    expect(wrapper.state().showNav).toEqual(true)
  })

  it('Should call showSidebar on click of hamburger and update state', () => {
    expect(wrapper.state().showNav).toEqual(false)
    jest.spyOn(wrapper.instance(), 'showSidebar')
    wrapper.find('.hamburger-link').simulate('click')
    expect(wrapper.instance().showSidebar).toHaveBeenCalled()
    expect(wrapper.state().showNav).toEqual(true)
  })

  it('Should call showSidebar on click of the x and update state', () => {
    expect(wrapper.state().showNav).toEqual(false)
    jest.spyOn(wrapper.instance(), 'showSidebar')
    wrapper.find('.hamburger-link').simulate('click')
    expect(wrapper.instance().showSidebar).toHaveBeenCalled()
    expect(wrapper.state().showNav).toEqual(true)
  })

  it('Should toggle state when showSidebar is called', () => {
    expect(wrapper.state().showNav).toEqual(false)
    wrapper.instance().showSidebar()
    expect(wrapper.state().showNav).toEqual(true)
    wrapper.instance().showSidebar()
    wrapper.instance().showSidebar()
    wrapper.instance().showSidebar()
    expect(wrapper.state().showNav).toEqual(false)
    wrapper.instance().showSidebar()
    expect(wrapper.state().showNav).toEqual(true)
  })
}) 