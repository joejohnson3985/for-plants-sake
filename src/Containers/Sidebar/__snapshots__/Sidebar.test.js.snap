// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should match the snapshot when showNav is false 1`] = `
<div
  className="side-bar-container"
>
  <i
    className="fas fa-bars hamburger-link"
    onClick={[Function]}
  />
  <div
    className="bar"
  >
    <div>
      <i
        className="fas fa-times x-link"
        onClick={[Function]}
      />
      <h1>
        Peaking
      </h1>
      <NavLink
        activeClassName="current-nav"
        className="nav-link find-hikes-link"
        exact={true}
        onClick={[Function]}
        to="/search"
      >
        <h4>
          Find Hikes
        </h4>
      </NavLink>
      <NavLink
        activeClassName="current-nav"
        className="nav-link my-hikes-link"
        exact={true}
        onClick={[Function]}
        to="/my-hikes"
      >
        <h4>
          My Hikes
        </h4>
      </NavLink>
    </div>
  </div>
</div>
`;

exports[`Sidebar Should match the snapshot when showNav is true 1`] = `
<div
  className="side-bar-container"
>
  <i
    className="fas fa-bars hamburger-link"
    onClick={[Function]}
  />
  <div
    className="bar"
  >
    <div>
      <i
        className="fas fa-times x-link"
        onClick={[Function]}
      />
      <h1>
        Peaking
      </h1>
      <NavLink
        activeClassName="current-nav"
        className="nav-link find-hikes-link"
        exact={true}
        onClick={[Function]}
        to="/search"
      >
        <h4>
          Find Hikes
        </h4>
      </NavLink>
      <NavLink
        activeClassName="current-nav"
        className="nav-link my-hikes-link"
        exact={true}
        onClick={[Function]}
        to="/my-hikes"
      >
        <h4>
          My Hikes
        </h4>
      </NavLink>
    </div>
  </div>
</div>
`;
